aten/src/ATen/native/Blas.cpp

```diff
@@ -137,6 +140,110 @@ Tensor dot(const Tensor &self, const Tensor &other){
   });
 }
 
+// this is to port LDA_COND from the TH codebase
+// #define LDA_COND(M, N, LDA) ((N) == 1 || (LDA) >= THMax(1, (M)))
+// lda_cond function in this file is not used
+// because it requires `lda > std::max<int64_t>(1L, m)`
+// instead of `>=` as defined in the TH implementation
+constexpr inline bool lda_cond_addr(int64_t m, int64_t n, int64_t lda) {
+  return n == 1 || lda >= std::max<int64_t>(1L, m);
+}
+
+template<typename scalar_t>
+Tensor &addr_impl_cpu(Tensor &result, const Tensor &self,
+                    const Tensor& vec1, const Tensor& vec2,
+                    scalar_t beta, scalar_t alpha) {
+  if (&result != &self) {
+    at::native::resize_as_(result, self);
+    at::native::copy_(result, self);
+  }
+  if (beta == 0.0) {
+    at::native::zero_(result);
+  } else if (beta != 1.0) {
+    at::native::mul_(result, beta);
+  }
+
+  if (result.stride(0) == 1 &&
```

Comment:
This block needs a comment explaining it

Reply:
thanks for pointing this out, I will add a comment for explanations.